pkg/broker/container_manager.go

```diff
@@ -1,30 +1,18 @@
 package broker
 
 import (
-	"context"
-	"errors"
-
-	"code.cloudfoundry.org/lager"
 	"github.com/cloudfoundry-incubator/blockhead/pkg/config"
-	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
 	"github.com/pivotal-cf/brokerapi"
 )
 
-func provision(logger lager.Logger, client DockerClient, state *config.State, serviceDetails brokerapi.ProvisionDetails) (brokerapi.ProvisionedServiceSpec, error) {
-	service := state.Services[serviceDetails.ServiceID]
-	if service == nil {
-		return brokerapi.ProvisionedServiceSpec{}, errors.New("service not found")
-	}
-
-	plan := service.Plans[serviceDetails.PlanID]
-	if plan == nil {
-		return brokerapi.ProvisionedServiceSpec{}, errors.New("plan not found")
-	}
-
-	_, err := client.ImagePull(context.Background(), plan.Image, types.ImagePullOptions{})
-	if err != nil {
-		return brokerapi.ProvisionedServiceSpec{}, err
-	}
+//go:generate counterfeiter -o fakes/fake_docker_client.go . DockerClient
+type DockerClient interface {
+	client.ImageAPIClient
+	client.ContainerAPIClient
+}
 
-	return brokerapi.ProvisionedServiceSpec{}, nil
+//go:generate counterfeiter -o fakes/fake_container_manager.go . ContainerManager
+type ContainerManager interface {
+	Provision(*config.Service, *config.Plan) (brokerapi.ProvisionedServiceSpec, error)
```

Comment:
I think we were talking earlier about what this interface should be. I think provision is fine. I think we should change the brokerapi.ProvisionedServiceSpec if possible, just so that when we switch out the libraries we won't have to rewrite the container manager and their implementations, but that could be attacked on an another commit.

Reply:
yeah I totally agree. I wasnt sure what to return at this point as it very much depends on what we get for docker and kube but we can return a struct here that we then massage into a `brokerapi.ProvisionedServiceSpec` at the handlers layer. And when we swap brokerapi for something else, we only change stuff at the handlers layer.

Reply:
I like that idea.